libraries/ms-helper/inc/can_uart.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include "can_hw.h"
+#include "status.h"
+#include "uart.h"
+
+// Generic CAN HW <-> UART protocol
+// Requires CAN HW, UART to be initialized
+
+// Process RX'd CAN message from UART
+struct CanUart;
+typedef void (*CanUartRxCb)(const struct CanUart *can_uart, uint32_t id, bool extended,
+                            const uint64_t *data, size_t dlc, void *context);
+
+typedef struct CanUart {
+  UARTPort uart;
+  CanUartRxCb rx_cb;
+  void *context;
+} CanUart;
+
+// Module init: Note that the provided UART port will have its RX handler overwritten.
+StatusCode can_uart_init(CanUart *can_uart);
+
+// Expects an initialized module.
+// Overrides CAN HW's RX handler and passes requested TX's directly to CAN HW
+StatusCode can_uart_hook_can_hw(CanUart *can_uart);
+
+// Intended to request a TX on the receiver
```

Comment:
I looked at this comment and it really confused me until I looked at the implementation.

I think maybe it would be more clear to call this entire module a `CanUart{Proxy,Bridge,Converter,etc.}`? And a lot of these functions can probably be renamed to be more clear what they do.

Just to make sure I understand what's going on:

* `can_uart_init` initializes a UART RX handler that receives deserializes UART messages and either forwards it to the application (in the case of `CAN_UART_RX_MARKER`) or forwards it over CAN (in the case of `CAN_UART_TX_MARKER`)
* `can_uart_req_tx` serializes a received CAN message, and then forwards it via UART so that the receiving device replays it over CAN
* `can_uart_hook_can_hw`: receives CAN messages and then forwards them via UART, so that the receiving device receives it in the application layer

Reply:
Sure, I can try to come up with a better name.

No, the idea is that for non-slave devices, `can_uart` does not explicitly hook CAN with UART.

* ` can_uart_init` is correct. Although I'm not 100% sure always forwarding it to CAN is a good idea.
* `can_uart_req_tx` allows the application to request the slave device to TX a CAN message. This doesn't have to be a received CAN message, and in fact will likely never be one. For example, motor controller interface would use this to request drive command TX on the powertrain CAN bus.
* `can_uart_hook_can_hw` is correct.

Reply:
The intention behind separating the functions like this is so that a slave can pass messages between CAN and UART directly, without any transformation. On the other hand, a master wants to take messages from the main system CAN bus and convert them into messages that the devices on the slave bus will understand. Similarly, received messages from the slave bus that we'd like to forward must be translated to data that can be understood by the system CAN bus.

Reply:
@karlding - renamed `can_uart_hook_can_hw` to `can_uart_enable_passthrough` to reflect its intention a bit better. Let me know what you think.